### PR TITLE
feat: add ability to enable toc on certain posts ignore global toc settings

### PR DESCRIPTION
**Pull request by kirill-zak**
## What problem does this PR solve?

Add ability to enable toc on certain posts ignore global `ToC` settings.

## Is this PR adding a new feature?

Added higher priority for the post `ToC` parameter instead of the global `ToC` parameter.

## Is this PR related to any issue or discussion?

No

## PR Checklist

- [x] I have verified that the code works as described/as intended.
- [ ] This change adds a social icon which has a permissive license to use it.
- [x] This change **does not** include any external library/resources.
- [x] This change **does not** include any unrelated scripts (e.g. bash and python scripts).
- [ ] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

### CODE REVIEW

**Comment by kirill-zak** [comment]
Hi @hugo-sid! Can your review this PR?

File: exampleSite/content/en/posts/table-of-content/index.md [hunk 1]
```diff
@@ -19,6 +27,18 @@ To enable ToC on all posts (globally) set parameter `toc` to `true` in `config.t
...
+    ---
+    title: How to enable table of content
+    date: 2023-05-02
+    toc: true
+    ---
+    ```
+
 ## Disable table of content on certain posts
 
 To disable ToC on certain posts, you have to follow two steps.
```
Comment (Tai-ch0802) [inline exampleSite/content/en/posts/table-of-content/index.md:44]: nit: Suggest to modify additionally.
```md
To disable ToC on certain posts, you have to follow two steps.
Notice: `.Params.toc` in the post will overide `.Site.Params.toc`. After these steps, parameter `toc` will be `false`.
```

**Review by Tai-ch0802 — APPROVED** [review]
LGTM

File: exampleSite/content/en/posts/table-of-content/index.md (hunk 1 above)
Comment (kirill-zak) [reply exampleSite/content/en/posts/table-of-content/index.md:44]: Very well noted. I'll add it

**Review by hugo-sid — APPROVED** [review]
@kirill-zak sorry for the delay.

Thank you for improving ToC setting for Hugo blog awesome.

**Pull request merged**